### PR TITLE
[WIP] Add circleci jobs for php54

### PR DESCRIPTION
**Pull request by matt-thomason**
# Description
Adds circleci jobs for php54.

Fixes: https://app.asana.com/0/941895179897714/1147313559227492/f

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Test

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

**Comment by matt-thomason** [comment]
No need to review currently - still experimenting with this new php54 image.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,36 @@
 version: 2
 jobs:
+  unit-php54-mage18:
+    docker:
+      - image: italoc/php54
```
Comment (roopakv) [inline .circleci/config.yml]: is there an official image by circle ci? those load faster and we know what we run

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+      - image: circleci/mysql:5.7
```
Comment (roopakv) [inline .circleci/config.yml]: use the ram version

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+      - run:
+          name: PHP 5.4 Magento 1.8
+          command: |
+            mkdir ./artifacts
+            export TEST_ENV=php54-mage18
+            PHPUNIT_PHAR=tests/unit/phpunit-4.8.36.phar MAGENTO_VERSION=magento-mirror-1.8.1.0 tests/scripts/ci.sh
+      - store_artifacts:
+          path: ./artifacts
+
+  unit-php54-mage19:
```
Comment (roopakv) [inline .circleci/config.yml]: you can de-dupe the two above by making magento version a param

File: .circleci/config.yml (hunk 1 above)
Comment (matt-thomason) [reply .circleci/config.yml]: Unfortunately not - I'm trying a few images but they all have some issues. I might need to toy around with one of the ones that is closest.

File: .circleci/config.yml (hunk 3 above)
Comment (roopakv) [reply .circleci/config.yml]: @matt-thomason we might just want clean up this whole yml. If you are busy ignore this a d make this a tech debt

File: .circleci/config.yml (hunk 2 above)
Comment (roopakv) [reply .circleci/config.yml]: I think it's as simple as
```suggestion
      - image: circleci/mysql:5.7-ram
```

File: .circleci/config.yml (hunk 1 above)
Comment (roopakv) [reply .circleci/config.yml]: I see. Ok. Ignore this

File: .circleci/config.yml (hunk 3 above)
Comment (matt-thomason) [reply .circleci/config.yml]: Nah I'll make it right

**Comment by adentong** [comment]
@matt-thomason What's the status on this PR, since it's been opened for more than a month now? Can this closed or are you still going to be working on it?

**Comment by matt-thomason** [comment]
Yeah, I let this languish too long @adentong.
I need to rework some things - I'll close this for now and open a new one later this week.

**Pull request closed without merging**